packages/plugin-npm/tests/npmPublishUtils.test.js

```diff
@@ -0,0 +1,122 @@
+import {npath}      from '@yarnpkg/fslib';
+import {getGitHead} from '@yarnpkg/plugin-npm/sources/npmPublishUtils';
+import Tacks        from 'tacks';
+
+const File = Tacks.File;
+const Dir = Tacks.Dir;
+const fixtures = new Tacks(
+  Dir({
+    WithGitRepo: Dir({
+      ".git": Dir({
+        config: File(
+          `[core]\n` +
+          `\trepositoryformatversion = 0\n` +
+          `\tfilemode = false\n` +
+          `\tbare = false\n` +
+          `\tlogallrefupdates = true\n` +
+          `\tignorecase = true\n`
+        ),
```

Comment:
While comprehensive, those tests (using `tacks`) are quite non-idiomatic for our repository. I don't know what to do of it ... 🤔

Reply:
I think, rather than test `getGitHead` (which is very simple, has low risk, and probably isn't worth the test complexity), I'd prefer to instead inline the `execvp` call into the `publish` command and test `publish` itself.

We don't have tests covering `publish` at the moment because we have [our own mock server](https://github.com/yarnpkg/berry/blob/master/packages/acceptance-tests/pkg-tests-core/sources/utils/tests.ts#L216), and the publish endpoint hasn't been implemented yet. If you could add it, then we could simply write an integration test that would make sure that, when called, `yarn npm publish` submits a payload with a correct sha1 in `gitHead`.

Plus we'd then be able to start testing much more of the `publish` command, that would be a great improvement!

Reply:
- Got rid of Tacks.
- Tests are now testing the publish endpoint.